Mathlib/Data/Fin/Basic.lean

```diff
@@ -475,6 +475,14 @@ attribute [simp] castSucc_inj
 lemma castLE_injective (hmn : m ≤ n) : Injective (castLE hmn) :=
   fun _ _ hab ↦ Fin.ext (congr_arg val hab :)
 
+@[simp]
+lemma castLE_natCast {m n : ℕ} [NeZero m] (h : m ≤ n) (a : ℕ) :
+    letI : NeZero n := ⟨Nat.pos_iff_ne_zero.mp (lt_of_lt_of_le (Nat.pos_of_neZero m) h)⟩
```

Comment:
This looks to me like we should have the definition `[NeZero m] (h : m ≤ n)` implies `NeZero n`.

Reply:
Hmm, even if we can only ever use it to define local instances?

Reply:
I believe so yes, because that local instance can be useful multiple times, and so it has value as a lemma (in the same way as other lemmas are useful!). There's also precedent for such lemmas, such as [NeZero.of_pos](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/NeZero.html#NeZero.of_pos), [NeZero.of_map](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/GroupWithZero/Hom.html#NeZero.of_map) and [NeZero.of_injective](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/GroupWithZero/Hom.html#NeZero.of_injective).